pkg/types/aws/platform.go

```diff
@@ -144,6 +149,92 @@ type ServiceEndpoint struct {
 	URL string `json:"url"`
 }
 
+// VPCSpec configures the VPC for the cluster.
+type VPCSpec struct {
+	// Subnets defines the subnets in an existing VPC and can optionally specify their intended roles.
+	// If no roles are specified on any subnet, then the subnet roles are decided automatically.
+	// In this case, the VPC must not contain any subnets without the kubernetes.io/cluster/<cluster-id> tag.
+	//
+	// For manually specified subnet role selection, each subnet must have at least one assigned role,
+	// and the ClusterNode, IngressControllerLB, ControlPlaneExternalLB, and ControlPlaneInternalLB roles must be assigned to at least one subnet.
+	// However, if the cluster scope is internal, then ControlPlaneExternalLB is not required.
+	//
+	// Subnets must contain unique IDs, must not exceed a total of 40 subnets,
+	// and can include no more than 10 subnets with the IngressController role.
+	//
+	// Leave this field unset to have the installer create subnets in a new VPC on your behalf.
+	//
+	// Subnets is available in TechPreview.
```

Comment:
@patrickdillon  I remember vaguely that we didn't need to include a feature gate here. So is this line still valid (i.e. remove it)?

Reply:
No feature gate it is :D